### DISCUSSION TITLE
Size/length/count display

### DISCUSSION THREAD

**Discussion by vidhav** [directus/directus#13194] [Feature Requests]
What about a display component that shows the _size/length/count_ of whatever value is returned?

It could be as simple as running `value.length` (if that property exists) or `_.size(value)` (from Lodash).

**Comment by rijkvanzanten** [comment]
Would this be string length, or something like the number of related items?

**Reply by benhaynes** [reply]
Could be pretty cool! We should put some thought into making this as robust as possible. Maybe it has options to get the length of many things... as smartly as possible.

* String Length
* Character Count (Remove HTML / Markdown)
* Word Count
* Array Length

With options for "units" (singular and plural), conditional formatting (copied from other displays), and/or even a "target count" with the ability to show a progress bar. Just some random ideas. 😄

**Reply by rijkvanzanten** [reply]
Yeah! Just wanted to check if the OP was referring to just showing `X Items`, which is what the `related-values` display already does 🙂

**Reply by vidhav** [reply]
The idea was initially to have a display that could count anything with a length, @rijkvanzanten. I made a simple extension that I have mostly used for string and array lengths:

```javascript
export default {
  id: 'size',
  name: 'Size',
  component: function ({ value }) {
    return value.hasOwnProperty('length') ? value.length.toString() : '';
  },
  ...
}
```

It's also worth noting that the Lodash function "size" will count the keys of an Object.

The ideas and options you mention @benhaynes, sound really cool 👍

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
